tests/integration/test_envelope.py

```diff
@@ -12,3 +17,180 @@ def test_envelope(mini_sentry, relay_chain):
     event = mini_sentry.captured_events.get(timeout=1).get_event()
 
     assert event["logentry"] == {"formatted": "Hello, World!"}
+
+
+def generate_transaction_item():
+    return {
+        "event_id": "d2132d31b39445f1938d7e21b6bf0ec4",
+        "type": "transaction",
+        "transaction": "/organizations/:orgId/performance/:eventSlug/",
+        "start_timestamp": 1597976392.6542819,
+        "timestamp": 1597976400.6189718,
+        "contexts": {
+            "trace": {
+                "trace_id": "4C79F60C11214EB38604F4AE0781BFB2",
+                "span_id": "FA90FDEAD5F74052",
+                "type": "trace",
+            }
+        },
+        "spans": [
+            {
+                "description": "<OrganizationContext>",
+                "op": "react.mount",
+                "parent_span_id": "8f5a2b8768cafb4e",
+                "span_id": "bd429c44b67a3eb4",
+                "start_timestamp": 1597976393.4619668,
+                "timestamp": 1597976393.4718769,
+                "trace_id": "ff62a8b040f340bda5d830223def1d81",
+            }
+        ],
+    }
+
+
+def test_normalize_measurement_interface(mini_sentry, relay_chain):
+
+    # set up relay
+
+    relay = relay_chain()
+    mini_sentry.project_configs[42] = relay.basic_project_config()
+
+    # construct envelope
+
+    transaction_item = generate_transaction_item()
+
+    transaction_item.update(
+        {
+            "measurements": {
+                "LCP": {"value": 420.69},
+                "   lcp_final.element-Size  ": {"value": 1},
+                "fid": {"value": 2020},
+                "cls": {"value": None},
+                "fp": {"value": "im a first paint"},
+                "Total Blocking Time": {"value": 3.14159},
+            }
+        }
+    )
+
+    envelope = Envelope()
+    envelope.add_transaction(transaction_item)
+
+    # ingest envelope
+
+    relay.send_envelope(42, envelope)
+
+    envelope = mini_sentry.captured_events.get(timeout=1)
+
+    event = envelope.get_transaction_event()
+
+    # test actual output
+
+    assert event["transaction"] == "/organizations/:orgId/performance/:eventSlug/"
+    assert "trace" in event["contexts"]
+    assert "measurements" in event, event
+    assert len(event["measurements"]) == 3, event["measurements"]
+    # expect the key "LCP" to be lowercased as part of the normalization
+    assert event["measurements"]["lcp"]["value"] == 420.69
+    assert event["measurements"]["lcp_final.element-size"]["value"] == 1
+    assert event["measurements"]["fid"]["value"] == 2020
+    assert "cls" not in event["measurements"], event["measurements"]
+    assert "fp" not in event["measurements"], event["measurements"]
```

Comment:
What if you just assert the entire `event["measurements"] == { ... }`?

Reply:
Good point. Addressed in https://github.com/getsentry/relay/pull/724/commits/a3706abf8ba18820836d06977896178d03a546c1